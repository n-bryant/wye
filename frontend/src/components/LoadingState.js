import React from "react";
import PropTypes from "prop-types";

import createClassNameHelper from "@n_bryant/classnames-helper";
import JSS_CLASS_NAME_PREFIX from "../../lib/classNamePrefix";
import { withStyles } from "@material-ui/core/styles";

import CircularProgress from "@material-ui/core/CircularProgress";

import styles from "./LoadingState.styles";

/**
 * renders a loading state
 */
export const LoadingState = props => {
  const classnames = LoadingState.classnames(props);

  return (
    <div className={classnames.root()}>
      <CircularProgress className={classnames.element("progressIndicator")} />
    </div>
  );
};
LoadingState.classnames = createClassNameHelper(
  `${JSS_CLASS_NAME_PREFIX}LoadingState`
);
LoadingState.propTypes = {
  // styles to apply
  classes: PropTypes.shape({
    root: PropTypes.string,
    progressIndicator: PropTypes.string
  })
};
LoadingState.defaultProps = {
  classes: {}
};

export default withStyles(styles)(LoadingState);
