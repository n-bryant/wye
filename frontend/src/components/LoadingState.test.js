import { LoadingState } from "./LoadingState";

describe("LoadingState", () => {
  it("should render successfully", () => {
    const props = {
      classes: {
        root: "root",
        progressIndicator: "progressIndicator"
      }
    };
    expect(toJson(shallow(<LoadingState {...props} />))).toMatchSnapshot();
  });
});
