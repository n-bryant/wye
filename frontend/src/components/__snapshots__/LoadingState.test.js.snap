// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingState should render successfully 1`] = `
<div
  className="wye-LoadingState root"
>
  <WithStyles(ForwardRef(CircularProgress))
    className="wye-LoadingState__progressIndicator progressIndicator"
  />
</div>
`;
